File: Server/src/main/java/bgu/spl/net/impl/Frames/connectFrame.java
package bgu.spl.net.impl.Frames;

import java.util.ArrayList;
import java.util.Arrays;

public class connectFrame extends Frame implements clientFrame {

    private String acceptVersion;
    private String host;
    private String login;
    private String passcode;

    public connectFrame(String completeMsg) {
        super("CONNECT");
        ArrayList<String> tmp = new ArrayList<>();
        tmp.addAll(Arrays.asList(completeMsg.split("\n")));
        acceptVersion = getAfterChar(tmp.get(1), ':');
        host = getAfterChar(tmp.get(2), ':');
        login = getAfterChar(tmp.get(3), ':');
        passcode = getAfterChar(tmp.get(4), ':');
    }

    public String getAcceptVersion() {
        return acceptVersion;
    }

    public String getHost() {
        return host;
    }

    public String getLogin() {
        return login;
    }

    public String getPasscode() {
        return passcode;
    }

    @Override
    public String toString() {
        return getContent() + "ED" + "\n" + "accept-version:" + getAcceptVersion() + "\n\n^@"/*+ "\nhost:" + getHost() + "\nlogin:" + getLogin() + "\npasscode:" + getPasscode() + "\n"*/;
    }

    @Override
    public boolean isLegal(String s) {
        ArrayList<String> tmp = toArrayList(s);
        if (!tmp.get(1).startsWith("accept-version:")) return false;
        if (!tmp.get(2).startsWith("host:")) return false;
        if (!tmp.get(3).startsWith("login:")) return false;
        if (!tmp.get(4).startsWith("passcode:")) return false;
        return tmp.get(5).equals("");
    }
}

File: Client/src/connectFrame.cpp
//
// Created by rapkeb@wincs.cs.bgu.ac.il on 14/01/2020.
//

#include <iostream>
#include "../include/connectFrame.h"
using namespace std;

connectFrame::connectFrame(const string host, const string name, const string password):host(host),userName(name),password(password) {
}

string connectFrame::getHost() const {
    return this->host;
}

string connectFrame::getUserName() const {
    return this->userName;
}

string connectFrame::getPassword() const {
    return this->password;
}

string connectFrame::toString() {
    string tmp1 = "CONNECT";
    string tmp = tmp1 + "\n" + "accept-version:1.2\n" + "host:" + this->host + "\n" + "login:" + this->userName + "\n" + "passcode:" + this->password + "\n\n^@";
    return tmp;
}
